crates/oxc_linter/src/rules/vitest/require_local_test_context_for_concurrent_snapshots.rs

```diff
@@ -0,0 +1,173 @@
+use oxc_ast::{ast::MemberExpression, AstKind};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+
+use crate::{
+    context::LintContext,
+    rule::Rule,
+    utils::{
+        collect_possible_jest_call_node, is_type_of_jest_fn_call, JestFnKind, JestGeneralFnKind,
+        PossibleJestNode,
+    },
+};
+
+#[inline]
+fn is_snapshot_method(property_name: &str) -> bool {
+    matches!(
+        property_name,
+        "toMatchSnapshot"
+            | "toMatchInlineSnapshot"
+            | "toMatchFileSnapshot"
+            | "toThrowErrorMatchingSnapshot"
+            | "toThrowErrorMatchingInlineSnapshot"
+    )
+}
+
+#[inline]
+fn is_test_or_describe_node(member_expr: &MemberExpression) -> bool {
+    if let Some(id) = member_expr.object().get_identifier_reference() {
+        if matches!(
+            JestFnKind::from(id.name.as_str()),
+            JestFnKind::General(JestGeneralFnKind::Describe | JestGeneralFnKind::Test)
+        ) {
+            if let Some(property_name) = member_expr.static_property_name() {
+                return property_name == "concurrent";
+            }
+        }
+    }
+    false
+}
+
+fn require_local_test_context_for_concurrent_snapshots_diagnostic(span0: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Require local Test Context for concurrent snapshot tests")
+        .with_help("Use local Test Context instead")
+        .with_label(span0)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct RequireLocalTestContextForConcurrentSnapshots;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// The rule is intended to ensure that concurrent snapshot tests are executed within a properly configured local test context.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// test.concurrent('myLogic', () => {
+    ///     expect(true).toMatchSnapshot();
+    /// })
+    ///
+    /// describe.concurrent('something', () => {
+    ///     test('myLogic', () => {
+    ///         expect(true).toMatchInlineSnapshot();
+    ///     })
+    /// })
+    ///
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// test.concurrent('myLogic', ({ expect }) => {
+    ///     expect(true).toMatchSnapshot();
+    /// })
+    ///
+    /// test.concurrent('myLogic', (context) => {
+    ///     context.expect(true).toMatchSnapshot();
+    /// })
+    /// ```
+    RequireLocalTestContextForConcurrentSnapshots,
+    correctness,
+    suggestion
```

Comment:
Nit: we can remove this as it doesn't have a fix.

https://github.com/oxc-project/oxc/blob/a266b4516784eb03e4610b5dc55d0248dc9f554c/tasks/rulegen/template.txt#L37-L39